app/models/user.rb

```diff
@@ -100,7 +100,8 @@ class User < ActiveRecord::Base
   scope :idle,
     :conditions => ["lastvisit < ?", 30.minutes.ago.utc]
 
-  validates_uniqueness_of :username, :email, :steamid
+  validates_uniqueness_of :username, :email
+  validates_uniqueness_of :steamid, :allow_nil => true
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.